### PR TITLE
MINIFICPP-2016 Add session commit time metrics

### PR DESCRIPTION
**Pull request by lordgamez**
https://issues.apache.org/jira/browse/MINIFICPP-2016

--------------------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: libminifi/test/unit/MetricsTests.cpp [hunk 1]
```diff
@@ -246,6 +246,38 @@ TEST_CASE("Test ProcessorMetrics", "[ProcessorMetrics]") {
...
+
+  for (auto i = 0; i < 10; ++i) {
+    metrics.addLastSessionCommitRuntime(40ms);
+  }
+  REQUIRE(metrics.getAverageSessionCommitRuntime() == 40ms);
+  REQUIRE(metrics.getLastSessionCommitRuntime() == 40ms);
+
+  metrics.addLastSessionCommitRuntime(10ms);
+  REQUIRE(metrics.getLastSessionCommitRuntime() == 10ms);
+  REQUIRE(metrics.getAverageSessionCommitRuntime() == 37ms);
```
Comment (fgerlits) [inline libminifi/test/unit/MetricsTests.cpp:280]: I would put these in a separate `TEST_CASE`

File: libminifi/src/core/ProcessorMetrics.cpp [hunk 2]
```diff
@@ -105,6 +110,17 @@ std::chrono::milliseconds ProcessorMetrics::getLastOnTriggerRuntime() const {
...
+void ProcessorMetrics::addLastSessionCommitRuntime(std::chrono::milliseconds runtime) {
+  session_commit_runtime_averager_.addValue(runtime);
+}
+
+std::chrono::milliseconds ProcessorMetrics::getLastSessionCommitRuntime() const {
+  return session_commit_runtime_averager_.getLastValue();
+}
 
 template<typename ValueType>
 requires Summable<ValueType> && DividableByInteger<ValueType>
```
Comment (fgerlits) [inline libminifi/src/core/ProcessorMetrics.cpp:126]: at line 128, why is it safe to check `values_.empty()` outside of the lock?

File: libminifi/test/unit/MetricsTests.cpp (hunk 1 above)
Comment (lordgamez) [reply libminifi/test/unit/MetricsTests.cpp:287]: Updated in 7567374d9c44cf63b993eb76c1819ea957832e59

File: libminifi/src/core/ProcessorMetrics.cpp (hunk 2 above)
Comment (lordgamez) [reply libminifi/src/core/ProcessorMetrics.cpp:126]: Good catch, it was a mistake, updated in 7567374d9c44cf63b993eb76c1819ea957832e59

**Review by fgerlits — APPROVED** [review]

**Review by szaszm — APPROVED** [review]

**Review by martinzink — APPROVED** [review]

**Review by adamdebreceni — APPROVED** [review]

**Pull request closed without merging**